### ISSUE TITLE
Regression - Failing with no locale

### ISSUE THREAD

**Issue by hell03610** [ember-intl/ember-intl#475]
I have many integration tests of components that use ember-intl for translations.
I was using ember-intl 2.18.0 and I was not setting the locale in the tests, as we are not testing the translations, and ignoring the "missing translation" messages with a configuration in the tests.

However, when updating to latest version 2.24.0, the tests fail when trying to fetch the locale.
Failure message is

```
TypeError: Cannot read property 'length' of null
    at Class.lookup (default.js:69)
    at Class.lookup (intl.js:154)
    at Class.getValue (format-message.js:57)
    at Class.compute (-format-base.js:67)
```

The new `lookup` method (former `findTranslationByKey`) does not always return an array, but it returns a null when no `localeName` is passed as parameter and no inner `_locale` is set.

https://github.com/ember-intl/ember-intl/blob/dd2a90b2ccf94000a92394565d048c42089aef9b/addon/services/intl.js#L153

Would it be ok for `_localeWithDefault` to return an array in all cases, so the library works even if no locale is set?

- [x] I am on the [latest](https://github.com/jasonmit/ember-intl/releases/latest) ember-intl version
- [x] I have searched the [issues](https://github.com/jasonmit/ember-intl/issues) of this repo and believe that this is not a duplicate

### Environment

- Ember Version: 2.11
- Ember CLI Version: 2.11
- Ember Intl Version: 2.18 - 2.24
- Browser(s): not related
- Node Version: 6.5

**Comment by jasonmit** [comment]
For now set your locale, but I'll work on a fix today/tomorrow.

**Comment by elopez-ag** [comment]
No worries, we have so many tests I prefer to wait for the fix than setting the locale in all of them 😅  thx!

**Comment by hell03610** [comment]
No worries, we have so many tests I prefer to wait for the fix than setting the locale in all of them 😅 thx!

**Comment by jasonmit** [comment]
https://github.com/ember-intl/ember-intl/pull/484 resolves this, will be in the next patch release

**Comment by jasonmit** [comment]
Fixes in 2.24.2, thanks for pointing out the issue
